services/wallet/router/router_v2.go

```diff
@@ -189,6 +189,14 @@ func newSuggestedRoutesV2(
 	allRoutes := node.buildAllRoutesV2()
 	allRoutes = filterRoutesV2(allRoutes, amountIn, fromLockedAmount)
 
+	if len(allRoutes) > 0 {
```

Comment:
Are we sorting routes and candidates just to make test assertions more reliable or is there a requirement to sort? If it's because of the former, can we remove all the prod code added in this PR and only sort the expected & actual results in the test file?

Reply:
Sorting routes/paths should provide the same best route if there are more routes with the same cost. So that's one reason for adding sorting, this also helps solve test assertions.

> can we remove all the prod code added in this PR

I'm not sure which prod code are you exactly referring to in this PR?

Reply:
Prod code would be anything not in test files. I just had the impression we could fix the flakiness by sorting the results only in the test, thus keeping prod code unchanged. But if you think it's beneficial to sort then it's fine 👍🏼 Thanks @saledjenic

Reply:
Yes, that's what I thought as well at first, but in the example I shared here https://github.com/status-im/status-go/pull/5605#issuecomment-2255215560 you can see that 2 routes with 2 transactions each like this
```
route 1:
10 from mainnet
2 from optimism
```
and this
```
route 2:
2 from mainnet
10 from optimism
```
in theory, may have exactly the same fee, which means when we sort them we will always have the same route.

Reply:
Thanks for the review. :)

Reply:
I agree that sorting in prod is good to have, it will give idempotent results to the user as well